### PR TITLE
Adds QuerySourceMapping.Remove API

### PR DESCRIPTION
**Pull request by anpete**
Small change to add a Remove API to QuerySourceMapping.

### CODE REVIEW

File: Core/Clauses/QuerySourceMapping.cs [hunk 1]
```diff
@@ -76,5 +76,18 @@ public Expression GetExpression (IQuerySource querySource)
...
     }
+
+    public void RemoveMapping (IQuerySource querySource)
+    {
+      ArgumentUtility.CheckNotNull ("querySource", querySource);
+
+      if (!ContainsMapping (querySource))
+      {
+        throw new InvalidOperationException (
+          string.Format ("Query source ({0}) has not been associated with an expression, cannot remove its mapping.", querySource));
```
Comment (MichaelKetting) [inline Core/Clauses/QuerySourceMapping.cs]: Two more whitespaces here, please: Double-indentation for statement continuation.

File: Core/Clauses/QuerySourceMapping.cs [hunk 2]
```diff
@@ -76,5 +76,18 @@ public Expression GetExpression (IQuerySource querySource)
 
       return expression;
     }
+
+    public void RemoveMapping (IQuerySource querySource)
```
Comment (MichaelKetting) [inline Core/Clauses/QuerySourceMapping.cs]: Please move it up to ReplaceMapping so we keep the modification-methods grouped together.

**Review by MichaelKetting — CHANGES_REQUESTED** [review]
Hi Andrew! 

Thanks for the PR. Some minor formatting issues but the code looks good, otherwise :)

It would be great if you could also create a corresponding feature request in our issue tracker (https://www.re-motion.org/jira/projects/RMLNQ/summary) and use the ticket number in the commit message (RMLNQ-xxx ...)

Do you have the Entity Framework feature at hand that needs the remove-logic? I'd like to learn more :)

~Michael

PS: We'll also need the CLA, please, before merging the request.

**Comment by anpete** [comment]
Created https://www.re-motion.org/jira/browse/RMLNQ-106

**Comment by anpete** [comment]
Updated.

As for the use case, it is basically that in our relational base framework we can end up removing an IQuerySource from a QueryModel _after_ a query mapping has been added for it by our core layer.

**Comment by MichaelKetting** [comment]
Thanks. Code looks good :) Commit message: Could you please put the issue number in front? Like this one here for example: https://github.com/re-motion/Relinq/commit/c01dbc83f048ed10586bdea05f00d282862cbed9

Re the use case: Ah, I see. Basically, a consistency guard to make sure nothing gets overlooked. Cool.

**Comment by anpete** [comment]
Commit message updated. I sent the CLA yesterday.

**Comment by MichaelKetting** [comment]
@anpete  I'll be doing a 2.2.0-alpha.1 release today or tomorrow but I'll wait with the next full release until there's a downstream stable release that depends on it :)

**Comment by anpete** [comment]
@MichaelKetting Thanks. Any chance we could get this one in soon? https://www.re-motion.org/jira/browse/RMLNQ-105

**Comment by MichaelKetting** [comment]
@anpete re rmlnq-105 I've put it on my shortlist and get back to you on this after this weekend. It's a bit of tricky issue, if I'm remembering this correctly.

**Comment by MichaelKetting** [comment]
@anpete re rmlnq-105: Okay, turned out to be tricky. I've allocated the upcoming Wednesday for it.

**Pull request merged**